### PR TITLE
Support for ConceptClass.

### PR DESCRIPTION
**Pull request by brandones**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/initializer/Domain.java [hunk 1]
```diff
@@ -4,17 +4,17 @@
...
+	                                        "programworkflowstates"), PERSON_ATTRIBUTE_TYPES(15,
+	                                                "personattributetypes"), IDENTIFIER_SOURCES(16, "idgen"), DRUGS(17,
+	                                                        "drugs"), ORDER_FREQUENCIES(18, "orderfrequencies"), ORDER_TYPES(
+	                                                                19, "ordertypes"), APPOINTMENTS_SPECIALITIES(20,
 	                                                                        "appointmentsspecialities"), APPOINTMENTS_SERVICES_DEFINITIONS(
-	                                                                                20,
+	                                                                                21,
 	                                                                                "appointmentsservicesdefinitions"), DATAFILTER_MAPPINGS(
-	                                                                                        21, "datafiltermappings");
+	                                                                                        22, "datafiltermappings");
```
Comment (brandones) [inline api/src/main/java/org/openmrs/module/initializer/Domain.java]: This file makes my eyes hurt

Comment (mks-d) [reply api/src/main/java/org/openmrs/module/initializer/Domain.java]: Lol, mine too, this freaking autoformatter is getting long enums _wrong_...

File: api-2.3/src/test/java/org/openmrs/module/initializer/api/loaders/LoadersIntegrationTest.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 import org.openmrs.module.initializer.api.appt.specialities.AppointmentsSpecialitiesLoader;
 import org.openmrs.module.initializer.api.attributes.types.AttributeTypesLoader;
 import org.openmrs.module.initializer.api.c.ConceptsLoader;
+import org.openmrs.module.initializer.api.c_class.ConceptClassLoader;
```
Comment (mks-d) [inline api-2.3/src/test/java/org/openmrs/module/initializer/api/loaders/LoadersIntegrationTest.java]: I think that'd be an unconventional package name, let's just do
```java
org.openmrs.module.initializer.api.c.ConceptClassLoader
```

File: api/src/main/java/org/openmrs/module/initializer/api/c_class/ConceptClassCsvParser.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	public Domain getDomain() {
+		return Domain.CONCEPT_CLASSES;
+	}
+	
+	@Override
+	public ConceptClass bootstrap(CsvLine line) throws IllegalArgumentException {
+		
+		String uuid = line.getUuid();
+		
+		ConceptClass c_class = service.getConceptClassByUuid(uuid);
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/module/initializer/api/c_class/ConceptClassCsvParser.java]: Ok so same here, that wouldn't be Java naming. Let's just be verbose: `conceptClass`.

File: api/src/main/java/org/openmrs/module/initializer/api/c_class/ConceptClassLineProcessor.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.openmrs.module.initializer.api.BaseLineProcessor;
+import org.openmrs.module.initializer.api.CsvLine;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ConceptClassLineProcessor extends BaseLineProcessor<ConceptClass> {
+	
+	protected static String HEADER_NAME = "name";
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/module/initializer/api/c_class/ConceptClassLineProcessor.java]: I will eventually revisit entirely the system of providing CSV header names, in the meantime I'm sure there was already a `HEADER_NAME` somewhere that can be imported.

The should all be `final public static` tho, also, in the meantime.

File: api/src/main/java/org/openmrs/module/initializer/api/c_class/ConceptClassLineProcessor.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	
+	@Autowired
+	public ConceptClassLineProcessor(@Qualifier("conceptService") ConceptService conceptService) {
+		super();
+		this.service = conceptService;
+	}
+	
+	public ConceptClass fill(ConceptClass c_class, CsvLine line) throws IllegalArgumentException {
+		
+		c_class.setName(line.get(HEADER_NAME, true));
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/module/initializer/api/c_class/ConceptClassLineProcessor.java]: We need the description too.

File: api/src/test/java/org/openmrs/module/initializer/api/ConceptClassLoaderIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+		}
+		{ // retired Procedure
+			ConceptClass c = service.getConceptClassByName("Procedure");
+			Assert.assertNotNull(c);
+			Assert.assertTrue(c.getRetired());
+		}
+		{ // created without uuid
+			ConceptClass c = service.getConceptClassByName("Animal");
+			Assert.assertNotNull(c);
+		}
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/module/initializer/api/ConceptClassLoaderIntegrationTest.java]: You should show that an existing class can be edited.

**Pull request merged**